cmd/kube-proxy/app/server.go

```diff
@@ -299,8 +299,19 @@ func (s *ProxyServer) Run() error {
 	// Tune conntrack, if requested
 	if s.Conntracker != nil {
 		if s.Config.ConntrackMax > 0 {
-			if err := s.Conntracker.SetMax(int(s.Config.ConntrackMax)); err != nil {
-				return err
+			err := s.Conntracker.SetMax(int(s.Config.ConntrackMax))
+			if err != nil {
+				if err != readOnlySysFSError {
+					return err
+				}
+				// readOnlySysFSError is caused by a known docker issue (https://github.com/docker/docker/issues/24000),
+				// the only remediation we know is to restart the docker daemon.
+				// Here we'll send an node event with specific reason and message, the
+				// administrator should decide whether and how to handle this issue,
+				// whether to drain the node and restart docker.
+				// TODO(random-liu): Remove this when the docker bug is fixed.
+				const message = "RESTART DOCKER TO REMEDY. Kube-proxy can't apply system configuration: /sys is read-only (docker issue #24000)."
```

Comment:
It's not the case that networking won't work properly.  I just might break
down sooner (run out of conntrack entries).  BUt yeah, OK.

"DOCKER RESTART NEEDED (docker issue #24000): /sys is read-only: can't
raise conntrack limits, problems may arise later"

On Sun, Jul 10, 2016 at 10:02 PM, David Oppenheimer <
notifications@github.com> wrote:

> In cmd/kube-proxy/app/server.go
> <https://github.com/kubernetes/kubernetes/pull/28697#discussion_r70202551>
> :
>
> > @@ -299,8 +299,19 @@ func (s *ProxyServer) Run() error {
> >  	// Tune conntrack, if requested
> >  	if s.Conntracker != nil {
> >  		if s.Config.ConntrackMax > 0 {
> > -			if err := s.Conntracker.SetMax(int(s.Config.ConntrackMax)); err != nil {
> > -				return err
> > +			err := s.Conntracker.SetMax(int(s.Config.ConntrackMax))
> > +			if err != nil {
> > +				if err != readOnlySysFSError {
> > +					return err
> > +				}
> > +				// readOnlySysFSError is caused by a known docker issue (https://github.com/docker/docker/issues/24000),
> > +				// the only remediation we know is to restart the docker daemon.
> > +				// Here we'll send an node event with specific reason and message, the
> > +				// administrator should decide whether and how to handle this issue,
> > +				// whether to drain the node and restart docker.
> > +				// TODO(random-liu): Remove this when the docker bug is fixed.
> > +				const message = "RESTART DOCKER TO REMEDY. Kube-proxy can't apply system configuration: /sys is read-only (docker issue #24000)."
>
> This message is missing why the user should care. I would suggest saying
> "Networking will not work properly on this node." in between those two
> sentences.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28697/files/12630bcd781ef0cc2cdec3adcf9e879fc227e1c4#r70202551>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFVgVBEJOUi7e4R7dvUqnh0bkDVUWHKEks5qUc6CgaJpZM4JIQlb>
> .
>